proto/rrg/winreg.proto

```diff
@@ -0,0 +1,59 @@
+// Copyright 2023 Google LLC
+//
+// Use of this source code is governed by an MIT-style license that can be found
+// in the LICENSE file or at https://opensource.org/licenses/MIT.
+syntax = "proto3";
+
+package rrg.winreg;
+
+// [Predefined key][1] of the Windows registry.
+//
+// Note that the integer representation **does not** correspond to the `HKEY_*`
+// constants as defined in the [`winreg.h`] header (they are out of the allowed
+// range for Protocol Buffer enums).
+//
+// [1]: https://learn.microsoft.com/en-us/windows/win32/sysinfo/predefined-keys
+enum PredefinedKey {
+    UNKNOWN = 0;
+    CLASSES_ROOT = 1;
+    CURRENT_USER = 2;
+    LOCAL_MACHINE = 3;
+    USERS = 4;
+    PERFORMANCE_DATA = 5;
+    CURRENT_CONFIG = 6;
+    PERFORMANCE_TEXT = 8;
```

Comment:
Is the 7 missing here on purpose?

Reply:
Sort of no (but not fully). Initially there was one more value (`DYN_DATA`) but it is no longer supported in Windows NT and the `winreg` crate won't support it anyway, so I deleted it. I considered to put `reserved` for its value since the initial idea was to use the same values as `HKEY_*` constants from the `winreg.h` header but that turned out to be infeasible (out of range for Protocol Buffers). Since the constants are sequential and start at 0x80000000, one could still argue for keeping some 1:1 mapping (it would be enough to add 0x80000000) which is why I skipped 7, but then there is `UNKNOWN` option anyway that makes the mapping even more questionable and make it is better to not attempt to replicate it in any form.

So, I re-assigned values sequentially.